commcare_connect/opportunity/tasks.py

```diff
@@ -357,3 +358,15 @@ def generate_catchment_area_export(opportunity_id: int, export_format: str):
     export_tmp_name = f"{now().isoformat()}_{opportunity.name}_catchment_area.{export_format}"
     save_export(dataset, export_tmp_name, export_format)
     return export_tmp_name
+
+
+@celery_app.task()
+def bulk_update_payment_accrued(opportunity_id, user_ids: list):
+    """Updates payment accrued for completed and approved CompletedWork instances."""
+    access_objects = OpportunityAccess.objects.filter(opportunity=opportunity_id, user__in=user_ids, suspended=False)
+    for access in access_objects:
+        with cache.lock(f"update_payment_accrued_lock_{access.id}", timeout=900):
+            completed_works = access.completedwork_set.exclude(status=CompletedWorkStatus.rejected).select_related(
```

Comment:
is this modeled off another query that does the same?

Reply:
yes, this is the same as the `update_payment_accrued` function.